Forms/Main.cs

```diff
@@ -1436,6 +1442,80 @@ private ListViewItem DownloadListItem(Model.Download info, ListViewItem item)
             return item;
         }
 
+        private void DownloadList_AddSmartName(int epid)
+        {
+            if (this.IsDisposed)
+            {
+                return;
+            }
+
+            if (this.InvokeRequired)
+            {
+                this.Invoke((MethodInvoker)(() => { this.DownloadList_AddSmartName(epid); }));
+                return;
+            }
+
+            ListViewItem item = this.ListDownloads.Items[Convert.ToString(epid, CultureInfo.InvariantCulture)];
+
+            if (item == null)
+            {
+                return;
+            }
+
+            if (this.downloadColOrder.Contains(Model.Download.DownloadCols.SmartName))
+            {
+                if (string.IsNullOrEmpty(item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.SmartName)].Text))
+                {
+                    string progName = item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.ProgrammeName)].Text;
+                    string epName = item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.EpisodeName)].Text;
+                    DateTime epDate = Convert.ToDateTime(item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.EpisodeDate)], CultureInfo.InvariantCulture);
+                    string smartName = TextUtils.BuildEpisodeSmartName(progName, epName, epDate);
+
+                    if (!string.IsNullOrEmpty(smartName))
+                    {
+                        item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.SmartName)].Text = smartName;
+                    }
+                }
+            }
+        }
+
+        private string DownloadList_GetSmartName(int epid)
+        {
+            if (this.IsDisposed)
+            {
+                return string.Empty;
+            }
+
+            if (this.InvokeRequired)
+            {
+                this.Invoke((MethodInvoker)(() => { this.DownloadList_GetSmartName(epid); }));
+                return string.Empty;
+            }
+
+            ListViewItem item = this.ListDownloads.Items[Convert.ToString(epid, CultureInfo.InvariantCulture)];
+
+            if (item == null)
+            {
+                return string.Empty;
+            }
+
+            if (this.downloadColOrder.Contains(Model.Download.DownloadCols.SmartName))
+            {
+                if (!string.IsNullOrEmpty(item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.SmartName)].Text))
+                {
+                    return item.SubItems[this.downloadColOrder.IndexOf(Model.Download.DownloadCols.SmartName)].Text;
+                }
+                else
+                {
+                    return string.Empty;
+                }
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
```

Comment:
If I understand correctly, neither of the proceeding functions are used, could you remove them?

Reply:
Yes your right, i had missed removing unused functions. Done.
Assuming that i should merge the 3 commits from your master before update this PR?

Reply:
Sure, go for it :smile: